### DISCUSSION TITLE
Run NLP on an ingredient “note”?

### DISCUSSION THREAD

**Discussion by jecorn** [mealie-recipes/mealie#2285] [Q&A] [answered]
I have a bunch of json recipes with the ingredients as “note” fields. Is there a way to run NLP on these during or after using the API to enter them into mealie? 

My json is certainly not quite complete, so maybe this already works and I just didn’t fully implement the schema. For example, I’m unclear if I need to set both the “disableAmount” and “quantity” fields when starting with a note-only recipe.

**Comment by michael-genson** [comment, marked as answer]
The `disableAmount` field is an indicator that, when flagged, tells Mealie to ignore everything other than the note field. This is how recipe parsers generally work (they extract the full ingredient as a single string of text and we put the result in the note)

So, if you're creating a recipe from a JSON file:
1. put the ingredients as a single string in the "note" field
2. set the quantity to 1 (if it's more/less than 1, put it in the note, don't change the quantity unless you're manually parsing the ingredient)
3. set `disableAmount` to `true` (this should be default)

If you're going to parse the ingredient notes using the NLP parser, set `disableAmount` to `false` and go through the parsing workflow in the UI (open edit mode and look for the button right under the ingredient list)

Mine is disabled because I already parsed this recipe, but it looks like this:
![image](https://user-images.githubusercontent.com/71845777/227580760-61806df4-b2e5-4b5b-a9df-6118f8ec002c.png)

**Reply by jecorn** [reply]
Thanks! Super clear. Is there a way to parse programmatically via the API? I have a very large number of recipes in json to import. No worries if not and I can use the method you posted. I’m just excited about the possibility of richer recipe data via the NLP.

**Reply by michael-genson** [reply]
Yup! There's an API endpoint for it. See the "Recipe: Ingredient Parser" section in your local docs (`https://your-mealie-url.com/docs`)

You might want to parse one through the UI to see how it works, the frontend does some extra things with that data to format it just right

**Reply by jecorn** [reply]
Oh! I looked right past it in the docs since I thought it would be under CRUD. Thanks for all your help!

**Reply by jecorn** [reply]
Follow up questions while I have your attention. I just saw that the parser api returns disableAmount:true. But if parsing is successful, should this be set to true? And confidence is returned as what looks like an empty dictionary rather than an example string/scalar. So what is the right key for the confidence dict?

**Reply by jecorn** [reply]
(I’m asking these questions rather than just trying it out because I’m on my phone and won’t get back to my mealie instance for a few hours. Happy to just give it a shot via the API in a bit if you’re out of steam with replies)

**Reply by michael-genson** [reply]
I believe the parser API just returns the current settings value for `disableAmount`, it doesn't change it, so you probably want to explicitly change it in your script.

Regarding the confidence, I'm not sure off the top of my head, but it's structured, so just try one request and check out the response
